### PR TITLE
feat(security): add proof-attempt audit producer

### LINKED ISSUE #3263: feat(security): proof-attempt audit producer MVP を追加する
> Language / 言語: 日本語 | English

## English Overview

Add a proof-attempt audit producer that evaluates each security claim against mapped code locations and emits candidate security findings.

## 背景

SPECA型監査の重要点は、単に「バグを探す」のではなく、「このsecurity claimが成立することを証明してみる。証明が破綻する場合のみcandidate findingにする」という framing にある。

ae-frameworkでは、このproof-attemptを security producer として扱い、出力を `security-finding/v1` に固定する。

## 目的

`security audit` CLIを追加し、security claim + code map から candidate finding を生成する。

## 提案CLI

```bash
ae security audit \
  --claims artifacts/security/security-claims.json \
  --code-map artifacts/security/security-code-map.json \
  --scope artifacts/security/security-audit-scope.json \
  --out artifacts/security/security-findings.json
```

## MVP方針

初期実装は2段階にする。

1. deterministic dry-run mode
   - claimごとに proof-attempt prompt input を生成する
   - findingは生成せず、audit task bundleを出力する

2. fixture-backed simulated mode
   - fixtureに用意した audit response を読み込み、`security-finding/v1` に正規化する

外部LLM実行は後続issueに分離する。

## 出力

```text
artifacts/security/security-audit-tasks.json
artifacts/security/security-findings.json
artifacts/security/security-audit-summary.md
```

## proof-attempt template

各claimについて、少なくとも次を構造化する。

```text
Map:
- Which code paths are relevant to this claim?

Prove:
- What must be true for the implementation to satisfy this claim?

Stress-Test:
- What attacker-controlled inputs or edge cases could break the proof?

Report:
- Only report a finding if the proof fails with concrete code evidence.
```

### In scope

- audit task bundle schema or internal type
- deterministic prompt/task generation
- fixture response normalization
- `security-finding/v1` 出力
- Markdown summary出力
- unit test

### Out of scope

- 外部LLM呼び出し
- exploit PoC生成
- vulnerability confirmation
- multi-worker orchestration
- budget/circuit breaker

## 受け入れ条件

- claim + code-mapから audit task bundle を生成できる
- fixture responseから `security-finding/v1` を生成できる
- findingは必ず `candidate` status で出力される
- `proofAttempt.map/prove/stressTest` がfinding内に残る
- no finding の場合もclaimごとの audit status がsummaryに出る
- outputがschema validationを通る

## Codex CLI 作業メモ

Codexには外部LLM統合をさせない。まずは「LLMに渡すべき構造化入力」と「返ってきた想定JSONの正規化」だけを実装する。

## ラベル案

- `enhancement`
- `security`
- `audit`
- `assurance`
- `codex`

---

## Import metadata

- Tracking roadmap: #3257
- Imported from local source: `tmp/ae-framework-security-issues/06-feat-proof-attempt-audit-producer-mvp.md`
- Imported by Codex from local issue definition files on 2026-05-07 JST.

### PR DESCRIPTION
**Pull request by ootakazuhiko**
## Summary

Closes #3263.

This PR adds the Security Assurance Lane proof-attempt audit producer MVP:

- adds `security-audit-task-bundle/v1` for deterministic proof-attempt prompt/task bundles
- adds `ae security audit` and `pnpm run security:proof-audit`
- supports dry-run task generation without external LLM calls
- supports fixture-backed response normalization into `security-finding/v1`
- keeps generated findings candidate-only and records no-finding status in the audit summary
- wires sample fixtures, schema validation, semantic validation, contract tests, and the Contract Catalog

## Acceptance

- `claim + code-map -> security-audit-task-bundle/v1` is implemented and schema-validated.
- Fixture-backed simulated responses produce `security-finding/v1`.
- Findings are normalized to `status: candidate`.
- `proofAttempt.map/prove/stressTest` remains in every generated finding.
- No-finding responses are represented in the Markdown audit summary by claim/task.
- No external LLM or network service is called by tests or the producer.

## Validation

- `pnpm -s exec vitest run tests/unit/security-assurance/proof-audit.test.ts tests/contracts/security-assurance-contract.test.ts --reporter dot`
- `pnpm -s run types:check`
- `pnpm -s run build`
- `node scripts/ci/validate-json.mjs`
- `pnpm -s run check:schemas`
- `pnpm -s run check:doc-consistency` (known pre-existing doc-governance warning only: `docs/quality/quality-implementation-status.md:158`)
- `pnpm -s run lint -- src/security/assurance/proof-audit.ts src/cli/security-cli.ts tests/unit/security-assurance/proof-audit.test.ts tests/contracts/security-assurance-contract.test.ts` (0 errors; existing warnings only)
- `git diff --check`

## Rollback

Revert this PR to remove the new proof-attempt audit producer, `security-audit-task-bundle/v1` schema/fixtures, and CLI wiring. Existing Security Assurance Lane schemas and producers from prior PRs remain independent.

### CODE REVIEW

File: src/security/assurance/proof-audit.ts [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+    outcome.result = 'finding';
+    const findingId = asString(finding['id']);
+    if (findingId !== undefined) {
+      outcome.findingIds.push(findingId);
+    }
+    outcomeByTaskId.set(task.id, outcome);
+  }
+  if (findings.length === 0) {
+    throw new Error('Security audit response fixture did not produce any security-finding/v1 findings; omit responseFixture for dry-run task generation only.');
+  }
```
[the remark this answers is not in the document]
Comment (ootakazuhiko) [reply src/security/assurance/proof-audit.ts]: 対応しました。`fix(security): allow no-finding audit responses` で、全responseが `no-finding` の場合も例外にせず、finding artifactは生成せずに task bundle と Markdown summary を出力する挙動に変更しました。claim/task別の `no-finding` status を summary に残す regression test も追加済みです。

検証:
- `pnpm -s exec vitest run tests/unit/security-assurance/proof-audit.test.ts tests/contracts/security-assurance-contract.test.ts --reporter dot`（24 tests pass）
- `pnpm -s run types:check`
- `pnpm -s run build`
- `node scripts/ci/validate-json.mjs`
- `git diff --check`
